app/scripts/models/auth_brokers/web-channel.js

```diff
@@ -96,14 +96,43 @@ define([
                 this, account, { closeWindow: true });
     },
 
+    beforeSignUpConfirmationPoll: function (account) {
+      // If the relier wants keys, the signup verification tab will need
+      // to be able to fetch them in order to complete the flow.
+      // Send them as part of the oauth session data.
+      if (this.relier.wantsKeys()) {
```

Comment:
Looking into this, I get the impression that:

* the password reset case should work already, because of the existing cross-tab dance that goes on to coordinate completion of that flow in the original tab
* the account-unlock flow only completes signin on the original tab, not the tab from the email link, and hence should work already.

If I can confirm that this is the case, I think it's better to special-case this to the sign-up flow for now, then deal with a more generalied version when we go to clean it up with e.g. tab-mutex or similar.